### PR TITLE
[WIP] Restructure Makefile to fix compile support

### PR DESCRIPTION
**Pull request by jopadan**
* reformat and fix build files
* update build instructions
* version bump v021

### CODE REVIEW

**Comment by Shoozza** [comment]
Hey thanks for the pull request.

There are a few unrelated changes mixed in there.
I hope you don't mind if I split your pull request into smaller stuff.
I've put the steam build error into another pull request as it should be easily to merge it.

The other changes will follow but need some discussion or changes.

STEAM:
Note the current soldat version on steam uses steam_api.dll v020.
And I'll have to look up what changed. I guess since this will take some time to finish we will have a few steam version changes so not sure if we just stay on v020 for now (or are there things that require us to update?)

MAKE via $(varnames)
I'm currently not sure if the changes for the makefile should be done in this way.
For example we only support windows 86_64 and i386. From the current make file that is clear. The changes might give people a wrong impression.

MAKE appid genreation + linux only fixes
The appid generation for steam builds probably should only happen when we build a steam version, i think.
It seems like there are some changes necessary for the windows build as well as this patch only includes linux/macOS part.

MAKE build help for GameNetworkingSockets:
I guess we probably should include the source code from GameNetworkingSockets in 3rdparty folder as version changes lately broke it, instead git cloing via makefile if missing. Seems like that would be doing too much in a make file.

MAKE uname fixes:
The uname changes are interesting, i think we need to fix a few assumptions besides those that you noticed and fixed.
https://en.wikipedia.org/wiki/Uname has a list of possible replies. Something something amd64 and i686.

Thanks for the work! It we take all those into account we get an easier build process.

**Comment by Shoozza** [comment]
Just reviewed the steamworks sdk changes between 1.48a and 1.49 the there are more changes than just the one change from v020 to v021 (diff the steam_api.json file). I think for now we will stay on 1.48a for soldat.

I'll create another repository for accepting the latest changes in the steamworks sdk and once it is stable we can update the soldat steamworks wrapper.

File: README.md [hunk 1]
```diff
@@ -33,9 +33,10 @@ Soldat compiles on Windows, Linux and macOS.
 
 ### Compilation using Makefiles
 
-0. Create build directory structure: `mkdir -p server/build/linux; mkdir -p client/build/linux`
-1. Compile options: make -e build ARCH=(x86_64|x86|arm) SYSTEM=(windows|linux|osx) SYSTEM_ARCH=(win64,win32,arm,linux,darwin) steam=(on|off) ARCH_EXT=(_x64|_arm|"")
+1. Compile options: make build -e ARCH=(x86_64|x86|arm) SYSTEM=(windows|linux|osx) \
+                    SYSTEM_ARCH=(win64,win32,arm,linux,darwin) steam=(on|off) \
+		    ARCH_EXT=(_x64|_arm|"")
```
Comment (furai) [inline README.md]: The indentation here on GitHub looks fishy, could you verify all is correct here? @Shoozza 

Also, isn't standard for new lines in MD 2 spaces at the end of the line? Maybe backslash is more compatible.
I've also seen `<br/>` being used. But I think that backslash is best option here cause it's verbose and short (unlike `<br/>`) and 2 spaces at the end will get trimmed by most of the modern editors.

**Review by furai — DISMISSED** [review]

File: README.md (hunk 1 above)
Comment (furai) [reply README.md]: Uh, nvm about wrong indent. After reload GitHub decided to show it completely different.

**Comment by BranDougherty** [comment]
Closed, the Makefiles are gone now.

**Pull request closed without merging**